File: app/src/main/java/kr/djspi/pipe01/fragment/PhotoDialog.java
package kr.djspi.pipe01.fragment;

import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import kr.djspi.pipe01.R;

import static kr.djspi.pipe01.Const.TAG_PHOTO;

public class PhotoDialog extends DialogFragment implements OnClickListener {

    private static final String TAG = PhotoDialog.class.getSimpleName();
    private int selectIndex = -1;
    private String dialogTitle;
    private OnSelectListener listener;

    public PhotoDialog() {
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnSelectListener) {
            listener = (OnSelectListener) context;
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        dialogTitle = getString(R.string.popup_title_select_photo);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_photo, container, false);

        TextView titleView = view.findViewById(R.id.popup_title);
        titleView.setText(dialogTitle);

        view.findViewById(R.id.btn_close).setOnClickListener(this);
        view.findViewById(R.id.btn_camera).setOnClickListener(this);
        view.findViewById(R.id.btn_gallery).setOnClickListener(this);

        return view;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_camera:
                selectIndex = 1;
                listener.onSelect(TAG_PHOTO, selectIndex, (String) null);
                dismissAllowingStateLoss();
                break;
            case R.id.btn_gallery:
                selectIndex = 2;
                listener.onSelect(TAG_PHOTO, selectIndex, (String) null);
                dismissAllowingStateLoss();
                break;
            case R.id.btn_close:
                dismissAllowingStateLoss();
            default:
                break;
        }
    }

    @Override
    public void onDismiss(DialogInterface dialog) {
        selectIndex = -1;
        super.onDismiss(dialog);
    }

    @Override
    public void onDetach() {
        super.onDetach();
        listener = null;
    }
}

File: app/src/main/java/kr/djspi/pipe01/fragment/PhotoDialog.kt
package kr.djspi.pipe01.fragment

import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import kr.djspi.pipe01.Const.TAG_PHOTO
import kr.djspi.pipe01.R

class PhotoDialog : DialogFragment(), View.OnClickListener {

    private var dialogTitle: String? = null
    private var selectIndex: Int = -1
    private lateinit var listener: OnSelectListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnSelectListener) {
            listener = context
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        dialogTitle = getString(R.string.popup_title_select_photo)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_photo, container, false)
        val title = view.findViewById<TextView>(R.id.popup_title)
        title.text = dialogTitle
        arrayOf<View>(
            view.findViewById(R.id.button_close),
            view.findViewById(R.id.btn_camera),
            view.findViewById(R.id.btn_gallery)
        ).forEach {
            it.setOnClickListener(this)
        }
        return view
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_camera -> {
                selectIndex = 1
                listener.onSelect(TAG_PHOTO, selectIndex, null)
                dismissAllowingStateLoss()
            }
            R.id.btn_gallery -> {
                selectIndex = 2
                listener.onSelect(TAG_PHOTO, selectIndex, null)
                dismissAllowingStateLoss()
            }
            R.id.button_close -> dismissAllowingStateLoss()
        }
    }

    override fun onDismiss(dialog: DialogInterface) {
        selectIndex = -1
        super.onDismiss(dialog)
    }
}
